Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/LoadTestDataDBServiceImpl.java

```diff
@@ -86,18 +87,30 @@ public boolean savePatient(Patient patient) throws LoadTestDataException {
         boolean actionResult;
         if (CollectionUtils.isNotEmpty(patient.getPersonnames())) {
             actionResult = patientDAO.saveTransaction(patient);
-
+            updateDocumentWith(patient.getPatientId());
             if (!actionResult) {
                 logDaoError("Patient basic-info");
             }
-        }
-        else {
+        } else {
             LOG.info("Patient basic-info is required");
             throw new LoadTestDataException("Patient basic-info is required when trying to save-patient.");
         }
         return actionResult;
     }
 
+    private void updateDocumentWith(long patientid) throws LoadTestDataException {
+        List<Document> updateDocument = documentDAO.findAllBy(patientid);
+        if (updateDocument != null) {
+            Patient patient = patientDAO.readTransaction(patientid, true);
+            for (int i = 0; i < updateDocument.size(); i++) {
+                HelperUtil.updateDocumentBy(updateDocument.get(i), patient);
+                saveDocument(updateDocument.get(i));
```

Comment:
there should be a documentDAO.saveAll with List<document> instead of saving-one-document at a time